test/LaunchDarkly.EventSource.Tests/EventSourceTests.cs

```diff
@@ -14,187 +14,95 @@ namespace LaunchDarkly.EventSource.Tests
-
-            for (int i = 0; i < 100; i++)
-            {
-
-                var backoff = expo.GetNextBackOff();
-
-                Assert.True(backoff <= max);
-            }
-
-        }
-
-        [Fact]
-        public void Exponential_backoff_should_reset_when_reconnect_count_resets()
-        {
-            TimeSpan max = TimeSpan.FromMilliseconds(30000);
-
-            ExponentialBackoffWithDecorrelation expo =
-                new ExponentialBackoffWithDecorrelation(TimeSpan.FromMilliseconds(1000), max);
-
-            for (int i = 0; i < 100; i++)
-            {
-                var backoff = expo.GetNextBackOff();
-            }
-            expo.ResetReconnectAttemptCount();
-            // Backoffs use jitter, so assert that the reset backoff time isn't more than double the minimum
-            Assert.True(expo.GetNextBackOff() <= TimeSpan.FromMilliseconds(2000));
-        }
 
         [Fact]
         public async Task When_a_comment_SSE_is_received_then_a_comment_event_is_raised()
         {
-            // Arrange
-            var commentSent = ":";
+            var commentSent = ": hello";
 
             var handler = new StubMessageHandler();
+            handler.QueueResponse(StubResponse.Ok().WithAction(StubStreamAction.Write(commentSent + "\n\n")));
 
-            handler.QueueStringResponse(commentSent);
-
-            var evt = new EventSource(new Configuration(_uri, handler, readTimeout:_defaultReadTimeout));
+            var evt = new EventSource(new Configuration(_uri, handler));
 
-            string commentReceived = string.Empty;
-            var wasCommentEventRaised = false;
+            string commentReceived = null;
             evt.CommentReceived += (_, e) =>
             {
                 commentReceived = e.Comment;
-                wasCommentEventRaised = true;
-
                 evt.Close();
             };
 
-            //// Act
             await evt.StartAsync();
 
-            //// Assert
             Assert.Equal(commentSent, commentReceived);
-            Assert.True(wasCommentEventRaised);
         }
 
         [Fact]
         public async Task When_a_data_only_message_SSE_is_received_then_a_message_event_is_raised()
         {
-            // Arrange
             var sse = "data: this is a test message\n\n";
             var sseData = "this is a test message";
 
             var handler = new StubMessageHandler();
-
-            handler.QueueStringResponse(sse);
-
-            var evt = new EventSource(new Configuration(_uri, handler, readTimeout:_defaultReadTimeout));
-
-            MessageEvent message = null;
-            var wasMessageReceivedEventRaised = false;
-            evt.MessageReceived += (_, e) =>
-            {
-                message = e.Message;
-                wasMessageReceivedEventRaised = true;
-
-                evt.Close();
-            };
-
-            //// Act
+            handler.QueueResponse(StubResponse.Ok().WithAction(StubStreamAction.Write(sse)));

```

Comment:
For an explanation of why I added this "StubResponse/StubStreamAction" stuff instead of the simpler "QueueStringResponse", see StubHandler.cs.